### PR TITLE
Add key to customer group

### LINKED ISSUE #1458: Add key to customer group
The `CustomerGroup` has a new `key` property with new update actions. We should add the new property and update actions to our SDK.

### PR DESCRIPTION
**Pull request by acbeni**
Closes #1458

### CODE REVIEW

File: commercetools-models/src/test/java/io/sphere/sdk/customergroups/commands/CustomerGroupDeleteCommandByIdIntegrationTest.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import static io.sphere.sdk.test.SphereTestUtils.randomKey;
 import static org.assertj.core.api.Assertions.*;
 
-public class CustomerGroupDeleteCommandIntegrationTest extends IntegrationTest {
+public class CustomerGroupDeleteCommandByIdIntegrationTest extends IntegrationTest {
```
Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/customergroups/commands/CustomerGroupDeleteCommandByIdIntegrationTest.java]: Good catch 👍

File: commercetools-models/src/test/java/io/sphere/sdk/customergroups/CustomerGroupFixtures.java [hunk 2]
```diff
@@ -13,7 +13,9 @@
 public class CustomerGroupFixtures {
     public static void withCustomerGroup(final BlockingSphereClient client, final Consumer<CustomerGroup> consumer) {
         final String name = randomString();
-        final CustomerGroup customerGroup = client.executeBlocking(CustomerGroupCreateCommand.of(name));
+        final String key = randomKey();
```
Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/customergroups/CustomerGroupFixtures.java]: 👍

File: commercetools-models/src/main/java/io/sphere/sdk/customergroups/CustomerGroup.java [hunk 3]
```diff
@@ -47,6 +51,10 @@
     String getName();
 
     @Override
+    @Nullable
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/customergroups/CustomerGroup.java]: Here we could already use our existing update action generator or am I wrong? Cristian added a `@HasUpdateAction` annotation and we should try to use it here.

File: commercetools-models/src/main/java/io/sphere/sdk/customergroups/CustomerGroup.java [hunk 4]
```diff
@@ -29,13 +32,14 @@
...
 @HasCreateCommand(javadocSummary = "Creates a new customer group", includeExamples = "io.sphere.sdk.customergroups.commands.CustomerGroupCreateCommandIntegrationTest#execution()", interfaceContents = "static CustomerGroupCreateCommand of(final String groupName) {\n" +
         "        return of(CustomerGroupDraft.of(groupName));\n" +
         "    }")
-@HasUpdateCommand(javadocSummary = "Updates a customer group.")
-@HasDeleteCommand(javadocSummary = "Deletes a customer group.", includeExamples = "io.sphere.sdk.customergroups.commands.CustomerGroupDeleteCommandIntegrationTest#execution()")
+@HasUpdateCommand(javadocSummary = "Updates a customer group.",updateWithKey = true)
+@HasDeleteCommand(javadocSummary = "Deletes a customer group.", includeExamples = "io.sphere.sdk.customergroups.commands.CustomerGroupDeleteCommandByIdIntegrationTest#execution()", deleteWithKey = true)
 @HasQueryModel
-public interface CustomerGroup extends Resource<CustomerGroup> {
+public interface CustomerGroup extends Resource<CustomerGroup>, WithKey {
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/customergroups/CustomerGroup.java]: Good catch! I always forget to add the `With...` interfaces, but they're quite useful.

File: commercetools-models/src/main/java/io/sphere/sdk/customergroups/CustomerGroup.java (hunk 3 above)
Comment (acbeni) [inline commercetools-models/src/main/java/io/sphere/sdk/customergroups/CustomerGroup.java]: Actually i added the @HasUpdateAction 😅 , but the pull request is still pending thats why i didn't use it yet.

Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/customergroups/CustomerGroup.java]: It's merged now ;-)

Comment (acbeni) [inline commercetools-models/src/main/java/io/sphere/sdk/customergroups/CustomerGroup.java]: Done 😄

**Pull request merged**